src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function infinityAdder(number) {
+    if (typeof number === 'number') {
+      sum += number;
+
+      return infinityAdder;
+    } else {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
```

Comment:
```suggestion
      return infinityAdder;
    }
    
      const result = sum;

      sum = 0;

      return result;
```
else is redundant here